eval(soc-code-classifier)

lrperkins opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, **failure to follow the guidelines below will result in the PR being closed automatically**. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access be granted. 🚨

**PLEASE READ THIS**:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject it since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, please note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑

### Eval name

"Job Title to SOC Title Classifier"

### Eval description

This evaluation involves a machine learning model trained to classify job titles into their relevant Standard Occupational Classification (SOC) title from the Bureau of Labor Statistics (BLS). The model uses historical job title data and associated SOC titles to accurately predict the SOC title for any given job title.

### What makes this a useful eval?

This evaluation is incredibly valuable because it opens up a wealth of data possibilities tied to job titles. By accurately classifying job titles into their relevant SOC titles, we can access and leverage related data from resources like ONET, BLS, and census data. This can be particularly useful in labor market analyses, economic research, HR analytics, and other fields. Moreover, with an accurate SOC title classification, we can study trends, make predictions, and generate insights about various occupations, which could be beneficial for job seekers, employers, policymakers, and researchers alike.


## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high-quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should

- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your YAML is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (<https://platform.openai.com/docs/usage-policies>).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the commits on the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgment

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and the high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access be granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [x] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
 {"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Metal Worker'"}], "ideal": " Metal Workers and Plastic Workers, All Other"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Bread Baker'"}], "ideal": " Bakers"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Malt Liquors Sales Supervisor'"}], "ideal": " First-Line Supervisors of Non-Retail Sales Workers"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Duck Driver'"}], "ideal": " Tour Guides and Escorts"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Architect Specialist'"}], "ideal": " Marine Engineers and Naval Architects"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Golf Course Ranger'"}], "ideal": " Amusement and Recreation Attendants"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Sewing Supervisor'"}], "ideal": " First-Line Supervisors of Production and Operating Workers"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Screener and Blender'"}], "ideal": " Mixing and Blending Machine Setters, Operators, and Tenders"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Field Marketing Representative'"}], "ideal": " Sales Engineers"}
{"input": [{"role": "system", "content": "You are an expert in Standard Occupation Code (SOC) labor classifications issued by the Bureau of Labor Statistics. When give a job title to classify, respond with the correct BLS Title classification"}, {"role": "user", "content": "What is the SOC Code for the job title Surveyor'"}], "ideal": " Surveyors"}
  ```
</details>

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request the following changes:

1. `54231` samples are too many. Kindly add the maximum of  `2000` samples to the dataset.
2. Kindly revert all the changes in the `.gitattributes` file. You don't need to make any changes to this file. If git lfs is properly installed, the required files will automatically be pushed as lfs files.
3. Add a proper description of the eval in the `.yaml` file.
4. This eval is using the `ModelGraded` evaluation method, but no spec file is specified. Kindly specify the appropriate spec file.

We would love to review the PR again after the suggested changes.

lrperkins commented:
1. Reduced sample to 1998
2. Reverted changes to .gitattributies
3. Added description to yaml file
4. Added modelgraded_spec: fact

Thanks for your help!

usama-openai reviewed (changes requested):
Thanks for implementing the requested changes. I would like to request one more change.

This labor classification is not global and is issued by the American Bureau of Labor Statistics, so you need to add this to the prompt. You can update the prompt to something like: `You are an expert in Standard Occupation Code (SOC) labor classifications issued by the USA Bureau of Labor Statistics`.

lrperkins commented:
I updated all the prompts, and added clarification to the soc_codes.yaml.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.

lrperkins commented:
Looking forward to it!

lrperkins commented:
I received access to gpt-4 temporarily, but now my access has been removed. Is this expected?

usama-openai commented:
> I received access to gpt-4 temporarily, but now my access has been removed. Is this expected?

Thank you for drawing this to our attention. Give me some time to look into this.

usama-openai commented:
It seems like no account is associated with email address [levirperkins@gmail.com](mailto:levirperkins@gmail.com). Can you verify that your OpenAI account is associated with this email address.

lrperkins commented:
Oh, its associated with levi.r.perkins@gmail.com, because that's what I use for display on google login, but its the same email address. I have a second email on my github, without the periods, that might work: levi@perkins.work. Otherwise, let me know what would be helpful.
